canonical-websites/snapcraft.io
# Logout on sso failure

# Summary

Fixes #1090
Fixes https://sentry.io/canonical-bm/snapcraftio/issues/683010199/?query=is:unresolved

Improve requests to get/recharge the macaroon
Improve exception handling
Use same pattern as other api services that we have

# QA

- Log into snapcraft.io.
- Change SSO password.
- Wait for snapcraft.io's macaroon to need a refresh.
- You should be redirected to `/logout`
